buidl/message.py

```diff
@@ -0,0 +1,254 @@
+    value = 0
+
+    # Convert address to a ScriptPubKey
+    script_pubkey = None
+    try:
+        script_pubkey = address_to_script_pubkey(address)
+    except:
+        raise ValueError("Invalid address")
+        
+    tx_out = TxOut(value,script_pubkey)
+    
+    # create transaction
+    version=0
+    tx_inputs = [tx_in]
+    tx_outputs = [tx_out]
+    locktime=0
+    network="mainnet"
+
+    # TODO: Should this always be True? What about for FULL BIP322 with p2pkh?
+    segwit=True
+
+    return Tx(version,tx_inputs,tx_outputs,locktime,network,segwit)
+
+###################################################################################
+#
+# From BIP322
+# The to_sign Tx is:
+#    nVersion = 0 or (FULL format only) as appropriate (e.g. 2, for time locks)
+#    nLockTime = 0 or (FULL format only) as appropriate (for time locks)
+#    vin[0].prevout.hash = to_spend.txid
+#    vin[0].prevout.n = 0
+#    vin[0].nSequence = 0 or (FULL format only) as appropriate (for time locks)
+#    vin[0].scriptWitness = message_signature
+#    vout[0].nValue = 0
+#    vout[0].scriptPubKey = OP_RETURN
+#
+def create_to_sign_tx(to_spend_tx_hash, sig_bytes=None):
+
+    if (sig_bytes and is_full_signature(sig_bytes)):
+        
+        sig_stream = io.BytesIO(sig_bytes)
+        to_sign = Tx.parse(sig_stream)
+        
+        if (len(to_sign.tx_ins) > 1):
+            # TODO: Implement Proof of Funds
+            raise NotImplemented("Not yet implemented proof of funds yet")
+        elif (len(to_sign.tx_ins) == 0):
+            raise ValueError("No transaction input")
+        elif (to_sign.tx_ins[0].prev_tx != to_spend_tx_hash):
+            raise ValueError("The to_sign transaction input's prevtx id does not equal the calculated to_spend transaction id")
+        elif (len(to_sign.tx_outs) != 1):
+            raise ValueError("to_sign does not have a single TxOutput")
+        elif (to_sign.tx_outs[0].amount != 0):
+            raise ValueError("Value is Non 0", to_sign.tx_outs[0].amount)
+        elif(to_sign.tx_outs[0].script_pubkey.commands != [106]):
+            raise ValueError("ScriptPubKey incorrect", to_sign.tx_outs[0].script_pubkey)
+        else:
+            return to_sign
+            
+    else:
+        # signature is either None or an encoded witness stack
+        
+        # Identifies the index of the output from the virtual to_spend tx to be "spent"
+        prevout_index = 0
+        sequence = 0
+        # TxInput identifies the single output from the to_spend Tx
+        tx_input = TxIn(to_spend_tx_hash,prevout_index,script_sig=None,sequence=sequence)
+    
+        value = 0
+        # OP Code 106 for OP_RETURN
+        commands = [106]
+        scriptPubKey = Script(commands)
+
+        tx_output = TxOut(value,scriptPubKey)
+        locktime=0
+        version=0
+        tx_inputs = [tx_input]
+        tx_outputs = [tx_output]
+        network="mainnet"
+        # is to_sign always a segwit?
+        segwit=True
```

Comment:
I have run into some issues with the current hard coded `segwit=True` that is set of the `to_sign` transaction. Currently, I am attempting to create a bip322 p2sh multisig.

@jimmysong is there a recommended way for me to test whether this to_sign transation should have it's segwit flag set to True or not. 

It is to do with the scriptPubKey on the tx_out of the to_sign transaction right?